### PR TITLE
CIV-6614 toggle-reading-from-yml-file

### PR DESCRIPTION
**Pull request by nagaraju-deekonda**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-6614

### Change description ###

toggle reading from application yml file

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/civil-service/values.aat.template.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ java:
     FEIGN_CLIENT_CONFIG_REMOTEREPOSITORYSERVICE_URL: http://${SERVICE_NAME}-camunda/engine-rest/
     SERVICE_REQUEST_UPDATE: http://${SERVICE_FQDN}-java/service-request-update
     SERVICE_REQUEST_UPDATE_CLAIM_ISSUED: http://${SERVICE_FQDN}-java/service-request-update-claim-issued
+    WA_TOGGLE: WA3.5
```
Comment (sampankumar) [inline charts/civil-service/values.aat.template.yaml:39]: This can be removed for now. As aat and prod will share same value

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimCallbackHandler.java [hunk 2]
```diff
@@ -180,7 +182,7 @@ public List<CaseEvent> handledEvents() {
     private CallbackResponse startClaim(CallbackParams callbackParams) {
         CaseData.CaseDataBuilder caseDataBuilder = callbackParams.getCaseData().toBuilder();
         caseDataBuilder.claimStarted(YES);
-
+        caseDataBuilder.featureToggleWA(toggleConfiguration.getFeatureToggle());
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimCallbackHandler.java:185]: I think this should be done in RespondToDefenceCallbackHandler and the spec one

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -62,6 +62,8 @@ serviceRequestClaimIssued:
     site_id: AAA7
     service: CIVIL
 
+wa:
+  feature-toggle: ${WA_TOGGLE:}
```
Comment (sampankumar) [inline src/main/resources/application.yaml:66]: you can add a default value as 3.5 (as thats the current released version)

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimCallbackHandler.java (hunk 2 above)
Comment (sampankumar) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimCallbackHandler.java:185]: You dont need this here

Comment (sampankumar) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimCallbackHandler.java:185]: Also please add this in DefendantClaimDetailsNotificationHandler.java as well

**Review by sampankumar — APPROVED** [review]

**Review by sampankumar — APPROVED** [review]

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -62,6 +62,8 @@ serviceRequestClaimIssued:
     site_id: AAA7
     service: CIVIL
 
+wa:
+  feature-toggle: WA3.5
```
Comment (sampankumar) [inline src/main/resources/application.yaml:66]: @nagaraju-deekonda You were using a variable here till yesterday, Why has this changed?

**Review by sampankumar — APPROVED** [review]

**Review by sampankumar — DISMISSED** [review]

**Review by HarryH96 — APPROVED** [review]

**Review by sankhajuria — DISMISSED** [review]

File: charts/civil-service/values.yaml [hunk 5]
```diff
@@ -63,6 +63,7 @@ java:
     DOCUMENT_MANAGEMENT_SECURED: true
     REST_TEMPLATE_TIMEOUT: 30000
     POLLING_EVENT_EMITTER_ENABLED: false
+    WA_TOGGLE: WA3.5
```
Comment (sampankumar) [inline charts/civil-service/values.yaml:66]: Change this to WA4

File: Jenkinsfile_CNP [hunk 6]
```diff
@@ -8,7 +8,7 @@ import uk.gov.hmcts.contino.GithubAPI
 def type = "java"
 def product = "civil"
 def component = "service"
-def ccdBranch = "master"
+def ccdBranch = "CIV-6614-toggle-reading-from-yml-file"
```
Comment (HarryH96) [inline Jenkinsfile_CNP:11]: Will need reverted before merge @sampankumar

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java [hunk 7]
```diff
@@ -79,6 +81,7 @@ public List<CaseEvent> handledEvents() {
 
     private CallbackResponse notifyRespondentSolicitorForClaimDetails(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
+        caseData.setFeatureToggleWA(toggleConfiguration.getFeatureToggle());
```
Comment (HarryH96) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java:84]: why is this required?

File: Jenkinsfile_CNP (hunk 6 above)
Comment (sampankumar) [reply Jenkinsfile_CNP:11]: Correct

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java (hunk 7 above)
Comment (sampankumar) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java:84]: If I remember correctly this was done so that we dont have to do migration, as the case could be in any state when this gets released. 
We wanted to make sure that whatever is the scenario where we publish data we set this flag.
I believe we might have to add this at some other places as well.

Comment (sampankumar) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java:84]: @nagaraju-deekonda we need to add the same code in CreateClaimCallbackHandler as well
